datafusion/datasource/src/file.rs

```diff
@@ -126,6 +129,13 @@ pub trait FileSource: Send + Sync {
         ))
     }
 
+    fn try_pushdown_projection(
```

Comment:
I think we need to add doc strings to this method to help people who want to implement it -- specifically:
1. What the return value is (aka if None, the projection isn't pushed down)
2. If Some is returned, what is the expectation on the returned plan (e.g. the output schema changes, etc )

This would also be a good place to add hints / links to structures like `SplitProjection` that might help implementing such a thing

Reply:
Good call, done!